src/main/java/com/photolocos/enterprise/dto/Photo.java

```diff
@@ -18,7 +18,7 @@
 
     private String type;
     private String fileType;
-    private String location;
+    private Location location;
```

Comment:
Instead of giving a Photo a String location attribute, Photo's location attribute is now a Location object. This keeps all location related information encapsulated within the Location [(SOLID Principles: Single-responsibility principle)](https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
Definitely, good idea.

Strings often indicate improper use of objects, where a class type is more flexible.  The internal details of the class can change, and the method signatures that use it don't have to change.